spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -0,0 +1,223 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.boot.BootstrapRegistry.InstanceSupplier;
+import org.springframework.boot.ConfigurableBootstrapContext;
+import org.springframework.boot.cloud.CloudPlatform;
+import org.springframework.boot.context.config.ConfigDataLocation;
+import org.springframework.boot.context.config.ConfigDataLocationNotFoundException;
+import org.springframework.boot.context.config.ConfigDataLocationResolver;
+import org.springframework.boot.context.config.ConfigDataLocationResolverContext;
+import org.springframework.boot.context.config.ConfigDataResourceNotFoundException;
+import org.springframework.boot.context.config.Profiles;
+import org.springframework.boot.context.properties.bind.BindHandler;
+import org.springframework.boot.context.properties.bind.Bindable;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.cloud.kubernetes.commons.KubernetesClientProperties;
+import org.springframework.cloud.kubernetes.commons.KubernetesNamespaceProvider;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Ryan Baxter
+ */
+public abstract class KubernetesConfigDataLocationResolver
+		implements ConfigDataLocationResolver<KubernetesConfigDataResource>, Ordered {
+
+	static final boolean RETRY_IS_PRESENT = ClassUtils.isPresent("org.springframework.retry.annotation.Retryable",
+			null);
+
+	/**
+	 * Prefix for Config Server imports.
+	 */
+	public static final String PREFIX = "kubernetes:";
+
+	private final Log log;
+
+	public KubernetesConfigDataLocationResolver(Log log) {
+		this.log = log;
+	}
+
+	protected String getPrefix() {
+		return PREFIX;
+	}
+
+	@Override
+	public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDataLocation location) {
+		if (!location.hasPrefix(getPrefix())) {
+			return false;
+		}
+		// TODO Could we use KubernetesClientPodUtils.isInsideKubernetes instead?
+		CloudPlatform platformProperty = context.getBinder().bind("spring.main.cloud-platform", CloudPlatform.class)
```

Comment:
We can simplify at least this: 

```
CloudPlatform platformProperty = context.getBinder().bind("spring.main.cloud-platform", CloudPlatform.class)
				.orElse(CloudPlatform.NONE);
```

via 

```
CloudPlatform.KUBERNETES.isEnforced(context.getBinder());
```

Reply:
what I have tried is make `KubernetesConfigDataLocationResolver` implement `EnvironmentAware`, the have:

```
	private Environment env;

	public void setEnvironment(Environment env) {
		this.env = env;
	}
```

and re-write `isResolvable` to : 

```
	@Override
	public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDataLocation location) {
		if (!location.hasPrefix(getPrefix())) {
			return false;
		}

		return CloudPlatform.KUBERNETES.isEnforced(context.getBinder()) || CloudPlatform.KUBERNETES.isDetected(env);
	}
```

all the tests pass... is this something we could consider Ryan? wdyt?

Reply:
Hmmm IDK if the Environment is properly initialized at this point.  I am going to check with the Boot team

Reply:
nvm, we can't. `env` is null other here. tests were working because of luck, I've re-written the code slightly and it started failing saying that `env` is null. pity, honestly :( the code could have been dar easier to write.

Reply:
Yeah the Boot team said its probably not good to use `EnvironmentAware` but we can just create a new `StandardEnvironment` and use `isDetected` since it just uses environment variables